readthedocs/rtd_tests/tests/test_sync_versions.py

```diff
@@ -152,6 +153,498 @@ def test_new_tag_update_inactive(self):
+        on the user repository, the RTD's ``stable`` is back.
+        """
+        version8 = Version.objects.create(
+            project=self.pip,
+            identifier='0.8.3',
+            verbose_name='0.8.3',
+            type=TAG,
+            active=False,
+            machine=False,
+        )
+        self.pip.update_stable_version()
+        current_stable = self.pip.get_stable_version()
+
+        # 0.8.3 is the current stable
+        self.assertEqual(
+            version8.identifier,
+            current_stable.identifier
+        )
+        self.assertTrue(current_stable.machine)
+
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+            ],
+            'tags': [
+                # User new stable
+                {
+                    'identifier': '1abc2def3',
+                    'verbose_name': 'stable',
+                },
+                {
+                    'identifier': '0.8.3',
+                    'verbose_name': '0.8.3',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        current_stable = self.pip.get_stable_version()
+        version_stable = self.pip.versions.get(slug='stable')
+        self.assertEqual(
+            current_stable.identifier,
+            version_stable.identifier
+        )
+
+        # Deleting the tag should return the RTD's stable
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+            ],
+            'tags': [
+                {
+                    'identifier': '0.8.3',
+                    'verbose_name': '0.8.3',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        # The version 8 should be the new stable.
+        # The stable isn't stuck with the previous commit
+        version8 = self.pip.versions.get(slug='0.8.3')
+        current_stable = self.pip.get_stable_version()
+        self.assertEqual(
+            version8.identifier,
+            current_stable.identifier,
+        )
+        self.assertNotEqual(
+            '1abc2def3',
+            current_stable.identifier,
+        )
+        self.assertTrue(current_stable.machine)
+
+    def test_normal_behavior_for_stable_after_deleting_user_defined_tag_2(self):
```

Comment:
We should mention the difference with the previous test case in the name. Also, `normal_behaviour_for_stable` is not representative. Maybe we could use somethine like: `test_machine_attr_when_user_define_stable_tag` or `test_machine_attr_when_user_define_stable_tag_and_delete_it`.

Similar for the previous test case.